src/Microsoft.TestPlatform.CommunicationUtilities/JsonDataSerializer.cs

```diff
@@ -68,7 +94,35 @@ private JsonDataSerializer()
     /// <returns>A <see cref="Message"/> instance.</returns>
     public Message DeserializeMessage(string rawMessage)
     {
-        return Deserialize<VersionedMessage>(s_serializer, rawMessage);
+        if (!DisableFastJson)
```

Comment:
[cosmetic] I would revert the if and remove the else part to reduce nesting.

Reply:
Inverted